### ISSUE TITLE
Sidebar link with Jenkins user name veriable 

### ISSUE THREAD

**Issue by kooyunmo** [jenkinsci/sidebar-link-plugin#26]
### Feature Request

Hi, thank you for the great useful plugin.

It seems that the sidebar plugin does not allow to use any environment variable for the URL (e.g., `https://example.com&user=$USER_NAME`). I wish to use the dynamic URL link containing the logged-in Jenkins user ID. Is there any way to do this?

**Comment by damianszczepanik** [comment]
You are right. I don't know if USER_NAME is known from context plugin. Maybe TokenMacro can be added as dependency to have support for some variable

**Comment by KalleOlaviNiemitalo** [comment]
The TokenMacro.expand, TokenMacro.expandAll, and TokenMacro.evaluate methods  take a `Run<?,?>` or `AbstractBuild<?, ?>` parameter and often assume that the value is not null.  This does not seem suitable for sidebar links of a job, agent, folder, or the main page.  I think a custom set of macros would be a better solution.
